Tasks/NugetPackager/NuGetPackager.ps1

```diff
@@ -44,6 +44,11 @@ if ($b_versionByBuild)
     Write-Verbose "Autoversion: Getting version number from build"
     ##Get Version from Build
     
+    if($Env:SYSTEM_HOSTTYPE -eq "release") {
+        Write-Error (Get-LocalizedString -Key "Autoversion: Getting version number from build option is not supported in releases")
```

Comment:
Although within a release we will populate BUILD_BUILDNUMBER (this will correspond to version of primary artifact used within the Release), versioning Nuget package corresponding to BUILD_BUILDNUMBER within a Release doesn't make sense. Release be run using same build version multiple times (e.g. re-deploying with same build, creating multiple releases using same build etc.). It would be better to use release name (RELEASE_RELEASENAME) instead. Or we could introduce a generic variable (SYSTEM_NAME?) and use that instead of task having to know the context in which it gets executed. Based on how important it is for customers to use this option within release, we can do the needful. Until then, wanted to avoid the task failing with a cryptic error when running within Release & making it easier to diagnose.